table/src/main/java/tech/ydb/table/settings/ExecuteScanQuerySettings.java

```diff
@@ -10,31 +10,39 @@
 
 public class ExecuteScanQuerySettings {
     private final YdbTable.ExecuteScanQueryRequest.Mode mode;
-    private final long timeoutNanos;
+    private final Duration timeout;
     private final YdbTable.QueryStatsCollection.Mode collectStats;
 
     public ExecuteScanQuerySettings(Builder b) {
-        this.timeoutNanos = b.timeoutNanos;
+        this.timeout = b.timeout;
         this.mode = b.mode;
         this.collectStats = b.collectStats;
     }
 
+    public Duration getTimeoutDuration() {
+        return timeout;
+    }
+
     public static Builder newBuilder() {
         return new Builder();
     }
 
     public static final class Builder {
-        private long timeoutNanos = Duration.ofSeconds(60).toNanos();
+        private Duration timeout = Duration.ofSeconds(60);
         private YdbTable.ExecuteScanQueryRequest.Mode mode = MODE_EXEC;
         private YdbTable.QueryStatsCollection.Mode collectStats = STATS_COLLECTION_NONE;
 
         public Builder timeout(long duration, TimeUnit unit) {
-            this.timeoutNanos = unit.toNanos(duration);
+            if (duration > 0) {
```

Comment:
If duration <= 0, we ignore it without error and use 60 sec default. Should probably use illegal argument exception. same for timeout(Duration duration) and ReadTableSettings